taichi/ui/utils/utils.h

```diff
@@ -0,0 +1,218 @@
+
+
+#pragma once
+
+#include <string>
+#include <iostream>
+#include <stdexcept>
+#include <stdio.h>
+#include <iostream>
+#include <fstream>
+#include <stdexcept>
+#include <algorithm>
+#include <chrono>
+#include <vector>
+#include <cstring>
+#include <cstdlib>
+#include <cstdint>
+#include <array>
+#include <optional>
+#include <set>
+#include <unordered_map>
+
+#ifdef _WIN64
+#include <aclapi.h>
+#include <dxgi1_2.h>
+#include <windows.h>
+#include <VersionHelpers.h>
+#define _USE_MATH_DEFINES
+#endif
+
+#ifdef _WIN64
+#define VK_USE_PLATFORM_WIN32_KHR 1
+#endif
+#include <volk.h>
+#include <GLFW/glfw3.h>
+
+#include <stdarg.h>
+
+#define GLM_FORCE_DEPTH_ZERO_TO_ONE
```

Comment:
Reverse depth is better (one to zero) for numerical precision, we should consider that

Reply:
OK. Will use reverse depth. But `GLM_FORCE_DEPTH_ZERO_TO_ONE` is still needed here so that glm uses [0,1] instead [-1,1], and then we reverse the depth by swapping near and far and changing the compare op.